models/tasks.js

```diff
@@ -88,24 +88,74 @@ const addDependency = async (data) => {
  *
  * @return {Promise<tasks|Array>}
  */
+
+const getBuiltTasks = async (tasksSnapshot) => {
+  const tasks = buildTasks(tasksSnapshot);
+  const promises = tasks.map(async (task) => fromFirestoreData(task));
+  const updatedTasks = await Promise.all(promises);
+  const taskPromises = updatedTasks.map(async (task) => {
+    task.status = TASK_STATUS[task.status.toUpperCase()] || task.status;
+    const taskId = task.id;
+    const dependencySnapshot = await dependencyModel.where("taskId", "==", taskId).get();
+    task.dependsOn = [];
+    dependencySnapshot.docs.forEach((doc) => {
+      const dependency = doc.get("dependsOn");
+      task.dependsOn.push(dependency);
+    });
+    return task;
+  });
+  const taskList = await Promise.all(taskPromises);
+  return taskList;
+};
+
+const fetchPaginatedTasks = async ({ status = "", size = TASK_SIZE, page, next, prev }) => {
+  try {
+    const initialQuery = status
+      ? tasksModel.where("status", "==", status).orderBy("title")
+      : tasksModel.orderBy("title");
+    let queryDoc = initialQuery;
+
+    if (prev) {
+      queryDoc = queryDoc.limitToLast(size);
+    } else {
+      queryDoc = queryDoc.limit(size);
+    }
+
+    if (page) {
+      const startAfter = size * page;
+      queryDoc = queryDoc.offset(startAfter);
+    } else if (next) {
+      const doc = await tasksModel.doc(next).get();
+      queryDoc = queryDoc.startAt(doc);
+    } else if (prev) {
+      const doc = await tasksModel.doc(prev).get();
+      queryDoc = queryDoc.endAt(doc);
+    }
+
+    const snapshot = await queryDoc.get();
+
+    const first = snapshot.docs[0];
+    const prevDoc = await initialQuery.endBefore(first).limitToLast(1).get();
+
+    const last = snapshot.docs[snapshot.docs.length - 1];
+    const nextDoc = await initialQuery.startAfter(last).limit(1).get();
+
```

Comment:
RE

Reply:
This is again doing something similar like I explained above.
last contains the reference to the final document in the range of tasks to be returned. Using which we get the immediately next document.